File: src/main/java/sk/pa3kc/ui/calls/KeyCallback.java
package sk.pa3kc.ui.calls;

import org.lwjgl.glfw.GLFW;
import org.lwjgl.glfw.GLFWKeyCallback;

public class KeyCallback extends GLFWKeyCallback {
    private boolean[] keys = new boolean[512];

    @Override
    public void invoke(long window, int key, int scancode, int action, int mods) {
        this.keys[key] = action != GLFW.GLFW_RELEASE;
    }

    public boolean isKeyPressed(int keyCode) {
        return this.keys[keyCode];
    }
}

File: src/main/kotlin/sk/pa3kc/ui/call/KeyCallback.kt
package sk.pa3kc.ui.call

import org.lwjgl.glfw.GLFW
import org.lwjgl.glfw.GLFWKeyCallback

class KeyCallback : GLFWKeyCallback() {
    private val keys = BooleanArray(512)

    override fun invoke(window: Long, key: Int, scancode: Int, action: Int, mods: Int) {
        this.keys[key] = action != GLFW.GLFW_RELEASE
    }

    fun isKeyPressed(keyCode: Int): Boolean = this.keys[keyCode]
}
